pkg/database/managed_key.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+// ManagedKey is an interface that allows for a realm to manage signing keys
+// for different purposes.
+type ManagedKey interface {
+	// GetKID returns the public key version string
+	GetKID() string
+	// ManagedKeyID returns the reference to the key ID in the KMS.
+	ManagedKeyID() string
+	// IsActive() returns true if this key is active
+	IsActive() bool
+
+	SetRealmID(id uint)
```

Comment:
I think this function should be a separate interface, since some managed keys will not be realm specific, right? (Like system keys?) Or do you plan to use a realmID == 0 for those cases like we do for audit logs?

Reply:
Introduced RealmManagedKey - moved SetRealmID there.

this opens up extending ManagedKey to token signing keys in a follow up.